todo_app/app.py

```diff
@@ -1,17 +1,40 @@
 from flask import Flask, render_template, request, redirect
 from operator import itemgetter
-import pytest, datetime
+import pytest, datetime, os, requests, json
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+from todo_app.user import User
 
 from todo_app.flask_config import Config
 from todo_app.data.todo_items import ToDoCard, ViewModel, get_todo_cards, move_todo_card, create_todo_card, delete_todo_card
 
+
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    github_client =  WebApplicationClient(os.environ.get('CLIENTID'))
+    github_redirect = github_client.prepare_request_uri("https://github.com/login/oauth/authorize")
+
+    return redirect(github_redirect) 
+
+@login_manager.user_loader
+def load_user(github_user):
+    
+    return User(github_user)
+
+
 def create_app():  
     app = Flask(__name__)
+    app.secret_key = os.getenv('SECRET_KEY')
+    app.config['LOGIN_DISABLED'] = os.environ.get('LOGIN_DISABLED', 'False').lower() in ['true', '1']
```

Comment:
Nice idea, but you always set the value of LOGIN_DISABLED here, even if you're sometimes setting it to false, but your code expects the value to be empty if it's not true! As a read only user I still saw the write page, and was able to submit things

Instead you could either:
* Adjust the checks to be more like: `if not app.config['LOGIN_DISABLED']:`
* Or only set the app config if it's true:
```suggestion
    if os.environ.get('LOGIN_DISABLED', 'False').lower() in ['true', '1']:
        app.config['LOGIN_DISABLED'] = True
```